src/textbooks/textbook-form/TextbookForm.jsx

```diff
@@ -0,0 +1,192 @@
+        onSubmit={onSubmit}
+        validateOnBlur
+        validateOnMount
+      >
+        {({
+          values, handleSubmit, isValid, dirty, setFieldValue,
+        }) => (
+          <>
+            <Form.Group size="sm" className="form-field">
+              <Form.Label size="sm" className="font-weight-bold form-main-label">
+                {intl.formatMessage(messages.tabTitleLabel)} *
+              </Form.Label>
+              <FormikControl
+                name="tab_title"
+                value={values.tab_title}
+                placeholder={intl.formatMessage(messages.tabTitlePlaceholder)}
+              />
+              <Form.Control.Feedback className="form-helper-text">
+                {intl.formatMessage(messages.tabTitleHelperText)}
+              </Form.Control.Feedback>
+            </Form.Group>
+            <FieldArray
+              name="chapters"
+              render={(arrayHelpers) => (
+                <>
+                  {Boolean(values?.chapters.length) && values.chapters.map(({ title, url }, index) => (
+                    <div className="form-chapters-fields" data-testid="form-chapters-fields">
+                      <Form.Group size="sm" className="form-field">
+                        <Form.Label size="sm" className="form-label font-weight-bold required">
+                          {intl.formatMessage(messages.chapterTitleLabel)} *
+                        </Form.Label>
+                        <FormikControl
+                          name={`chapters[${index}].title`}
+                          value={title}
+                          placeholder={intl.formatMessage(messages.chapterTitlePlaceholder, { value: index + 1 })}
+                        />
+                        <Form.Control.Feedback className="form-helper-text">
+                          {intl.formatMessage(messages.chapterTitleHelperText)}
+                        </Form.Control.Feedback>
+                      </Form.Group>
+                      <Form.Group size="sm" className="form-field">
+                        <div className="d-flex align-items-center mb-1">
+                          <Form.Label size="sm" className="font-weight-bold mb-0">
+                            {intl.formatMessage(messages.chapterUrlLabel)} *
+                          </Form.Label>
+                          <IconButtonWithTooltip
+                            size="sm"
+                            variant="primary"
+                            className="ml-auto field-icon-button"
+                            tooltipContent="Upload"
+                            src={UploadIcon}
+                            iconAs={Icon}
+                            data-testid="chapter-upload-button"
+                            onClick={() => {
+                              setCurrentTextbookIndex(index);
+                              openUploadModal();
+                            }}
+                          />
+                          <IconButtonWithTooltip
```

Comment:
[important]: Please add an `alt` prop for `IconButtonWithTooltip` (it is required)

Reply:
Fixed